tests/network/primary_pod_network.go

```diff
@@ -0,0 +1,146 @@
+	var virtClient kubecli.KubevirtClient
+
+	BeforeEach(func() {
+		var err error
+		virtClient, err = kubecli.GetKubevirtClient()
+		Expect(err).NotTo(HaveOccurred(), "Should successfully initialize an API client")
+	})
+
+	Describe("Status", func() {
+		AssertReportedIP := func(vmi *v1.VirtualMachineInstance) {
+			By("Getting pod of the VMI")
+			vmiPod := tests.GetRunningPodByVirtualMachineInstance(vmi, tests.NamespaceTestDefault)
+
+			By("Making sure IP reported on the VMI matches the one on the pod")
+			Expect(vmi.Status.Interfaces[0].IP).To(Equal(vmiPod.Status.PodIP))
+			Expect(vmi.Status.Interfaces[0].IPs).NotTo(BeEmpty())
+			Expect(vmi.Status.Interfaces[0].IPs[0]).To(Equal(vmiPod.Status.PodIP))
+		}
+
+		Context("VMI connected to the pod network using the default (implicit) binding", func() {
+			var vmi *v1.VirtualMachineInstance
+
+			BeforeEach(func() {
+				vmi = setupVMI(virtClient, vmiWithDefaultBinding())
+			})
+
+			AfterEach(func() {
+				cleanupVMI(virtClient, vmi)
+			})
+
+			It("should report PodIP as its own on interface status", func() { AssertReportedIP(vmi) })
+		})
+
+		Context("VMI connected to the pod network using bridge binding", func() {
+			var vmi *v1.VirtualMachineInstance
+
+			BeforeEach(func() {
+				vmi = setupVMI(virtClient, vmiWithBridgeBinding())
+			})
+
+			AfterEach(func() {
+				cleanupVMI(virtClient, vmi)
+			})
+
+			It("should report PodIP as its own on interface status", func() { AssertReportedIP(vmi) })
+		})
+
+		Context("VMI connected to the pod network using masquerade binding", func() {
+			var vmi *v1.VirtualMachineInstance
+
+			BeforeEach(func() {
+				vmi = setupVMI(virtClient, vmiWithMasqueradeBinding())
+			})
+
+			AfterEach(func() {
+				cleanupVMI(virtClient, vmi)
+			})
+
+			It("should report PodIP as its own on interface status", func() { AssertReportedIP(vmi) })
+		})
+	})
+})
+
+func setupVMI(virtClient kubecli.KubevirtClient, vmi *v1.VirtualMachineInstance) *v1.VirtualMachineInstance {
+	By("Creating the VMI")
+	var err error
+	vmi, err = virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Create(vmi)
+	Expect(err).NotTo(HaveOccurred(), "VMI should be successfully created")
+
+	By("Waiting until the VMI gets ready")
+	vmi = tests.WaitUntilVMIReady(vmi, tests.LoggedInAlpineExpecter)
+
+	return vmi
+}
+
+func cleanupVMI(virtClient kubecli.KubevirtClient, vmi *v1.VirtualMachineInstance) {
+	if vmi != nil {
+		By("Deleting the VMI")
+		Expect(virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Delete(vmi.GetName(), &metav1.DeleteOptions{})).To(Succeed())
+
+		By("Waiting for the VMI to be gone")
+		Eventually(func() error {
+			_, err := virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Get(vmi.GetName(), &metav1.GetOptions{})
+			return err
+		}, 2*time.Minute, time.Second).Should(And(HaveOccurred(), WithTransform(errors.IsNotFound, BeTrue())), "The VMI should be gone within the given timeout")
```

Comment:
nit: maybe change the `AND` to `SatisfyAll`, it is equivalent but a little more readable I think. 
Inspired from Ram's note here : 
https://github.com/kubevirt/kubevirt/pull/3479#discussion_r468534242

Reply:
Done